src/diffpy/utils/diffraction_objects.py

```diff
@@ -284,6 +278,20 @@ def __rtruediv__(self, other):
 
     @property
     def all_arrays(self):
+        """The 2D array containing `xarray` and `yarray` values.
+
+        Returns
+        -------
+        ndarray
+            The 2D array containing the `xarray` values in q, tth, d, and `yarray`.
```

Comment:
for lower cognitive load I would change to `The `shape(len(data), 4)` 2D array with columns containing the `yarray` (intensity) and the `xarray` values in q, tth, and d .` so they appear in the docstring in the same order as in the array columns

Reply:
Yes, great! superb. The example and this docstring makes is super clear.